solver/errdefs/context.go

```diff
@@ -14,7 +14,7 @@ func IsCanceled(ctx context.Context, err error) bool {
 		return true
 	}
 	// grpc does not set cancel correctly when stream gets cancelled and then Recv is called
-	if err != nil && context.Cause(ctx) == context.Canceled {
+	if err != nil && errors.Is(context.Cause(ctx), context.Canceled) {
```

Comment:
We're only interested in cancelled contexts, here, correct? (as they commonly go hand-in-hand with `DeadlineExceeded`); e.g. https://github.com/moby/moby/pull/46444